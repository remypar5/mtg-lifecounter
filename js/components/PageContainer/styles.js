import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#161616',
        flex: 1,
        flexDirection: 'row',
        alignItems: 'stretch',
        justifyContent: 'flex-end',
    },
});

export default styles;
