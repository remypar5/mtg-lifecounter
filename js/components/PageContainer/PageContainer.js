import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const PageContainer = ({ style, children }) => {
    const containerStyles = [
        styles.container,
    ];

    if (style) {
        containerStyles.push(style);
    }

    return (
        <View style={containerStyles}>
            {children}
        </View>
    );
};

PageContainer.propTypes = {
    children: PropTypes.node.isRequired,
    style: PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.number,
        PropTypes.arrayOf([
            PropTypes.object,
            PropTypes.number,
        ]),
    ]),
};

PageContainer.defaultProps = {
    style: null,
};

export default PageContainer;
