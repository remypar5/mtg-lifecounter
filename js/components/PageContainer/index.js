import PageContainer from './PageContainer';
import styles from './styles';

export default PageContainer;
export {
    styles,
};
