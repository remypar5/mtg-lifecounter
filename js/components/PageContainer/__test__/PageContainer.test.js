import React from 'react';
import { shallow } from 'enzyme';

import PageContainer from '../PageContainer';

describe('<PageContainer />', () => {
    it('renders without crashing', () => {
        const component = shallow(<PageContainer>Hallo</PageContainer>);
        expect(component).toMatchSnapshot();
    });

    it('accepts a style attribute', () => {
        const component = shallow(<PageContainer style={{ width: '50%' }}>Hallo</PageContainer>);
        expect(component).toMatchSnapshot();
    });
});
