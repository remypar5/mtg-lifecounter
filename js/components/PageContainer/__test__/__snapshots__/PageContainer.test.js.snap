// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> accepts a style attribute 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageContainer
    style={
      Object {
        "width": "50%",
      }
    }
  >
    Hallo
  </PageContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Hallo",
      "style": Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#161616",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        Object {
          "width": "50%",
        },
      ],
    },
    "ref": null,
    "rendered": "Hallo",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Hallo",
        "style": Array [
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#161616",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
          Object {
            "width": "50%",
          },
        ],
      },
      "ref": null,
      "rendered": "Hallo",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<PageContainer /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageContainer
    style={null}
  >
    Hallo
  </PageContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Hallo",
      "style": Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#161616",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
      ],
    },
    "ref": null,
    "rendered": "Hallo",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Hallo",
        "style": Array [
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#161616",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
        ],
      },
      "ref": null,
      "rendered": "Hallo",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
